### PR TITLE
Update to nx 19.3, angular 18, storybook 8

### PR DESCRIPTION
**Pull request by Markus-Ende**

### CODE REVIEW

File: libs/sketch/.storybook/main.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import type { StorybookConfig } from '@storybook/angular';
 
 const config: StorybookConfig = {
-  stories: ['../**/*.stories.@(js|jsx|ts|tsx|mdx)'],
+  stories: ['../**/*.@(mdx|stories.@(js|jsx|ts|tsx))'],
```
Comment (EeveeInGaa) [inline libs/sketch/.storybook/main.ts:4]: is there a reason you moved it or was it done automatically?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,15 +23,18 @@ jobs:
       # - run: npx nx-cloud start-ci-run --distribute-on="5 linux-medium-js" --stop-agents-after="e2e-ci"
 
       # Cache node_modules
-      - uses: actions/setup-node@v3
+      - uses: actions/setup-node@v4
         with:
-          node-version: 20
+          node-version-file: package.json
```
Comment (EeveeInGaa) [inline .github/workflows/ci.yml:28]: cool

File: .eslintrc.json [hunk 3]
```diff
@@ -201,7 +203,9 @@
         "no-var": "error",
         "prefer-const": "error",
         "prefer-template": "error",
-        "unused-imports/no-unused-imports-ts": 2
+        "unused-imports/no-unused-imports-ts": 2,
```
Comment (EeveeInGaa) [inline .eslintrc.json:206]: what does the 2 means? Is it a threshold or so?

File: libs/sketch/src/lib/components/dialog/dialog.component.stories.ts [hunk 4]
```diff
@@ -99,7 +97,7 @@ Default.argTypes = {
     description: 'the position of the close button',
     defaultValue: { summary: CloseButtonPosition.Right },
   },
-};
+} satisfies ArgTypes;
```
Comment (EeveeInGaa) [inline libs/sketch/src/lib/components/dialog/dialog.component.stories.ts:100]: is this to get rid of the as .. as ..? if yes this is great to know! I really didnt liked it. 
If not, what is it for?

File: migrations.json [hunk 5]
```diff
@@ -0,0 +1,72 @@
+{
```
Comment (EeveeInGaa) [inline migrations.json:1]: have you added it or got it added automatically?

**Review by EeveeInGaa — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Markus-Ende) [reply .github/workflows/ci.yml:28]: it reads the version from volta property

File: .eslintrc.json (hunk 3 above)
Comment (Markus-Ende) [reply .eslintrc.json:206]: I don't know, was automatically added by the update migration.

File: libs/sketch/.storybook/main.ts (hunk 1 above)
Comment (Markus-Ende) [reply libs/sketch/.storybook/main.ts:4]: was done by the storybook migration script. Something with the markdown files changed after update to v8

File: migrations.json (hunk 5 above)
Comment (Markus-Ende) [reply migrations.json:1]: Ah, this is generated by nx update migration. But could also be added to gitignore. But also does no harm when added

**Pull request merged**